TidepoolServiceKit/TidepoolService.swift

```diff
@@ -69,17 +71,25 @@ public final class TidepoolService: Service, TAPIObserver, ObservableObject {
     private let log = OSLog(category: "TidepoolService")
     private let tidepoolKitLog = OSLog(category: "TidepoolKit")
 
+    private var deviceLogUploader: DeviceLogUploader?
+
     public init(hostIdentifier: String, hostVersion: String) {
         self.id = UUID().uuidString
         self.hostIdentifier = hostIdentifier
         self.hostVersion = hostVersion
 
         Task {
-            await tapi.setLogging(self)
-            await tapi.addObserver(self)
+           await finishSetup()
         }
     }
 
+    public func finishSetup() async {
+        await tapi.setLogging(self)
+        await tapi.addObserver(self)
```

Comment:
Do these need to run sequentially or in this particular order? If not, we could run them simultaneously to get a tad bit more performance with a `TaskGroup`.

```suggestion
        await withTaskGroup { group in
            group.addTask { await tapi.setLogging(self) }
            group.addTask { await tapi.addObserver(self) }
            await group.waitForAll()
        }
```

Reply:
I'd be very surprised if this created any measurable performance improvement. Combining that with the additional code and complexity and loss of readability, I don't think it would be worth it, especially for a situation where performance isn't important in the first place.